### ISSUE TITLE
loopback saves data with invalid Date property

### ISSUE THREAD

**Issue by aantipov** [strongloop/loopback#1035]
Hi,
I use loopback version 2.10.2.
I have a model with a base "PersistedModel" and defined some properties on it.
```
{
  "name": "Hike",
  "base": "PersistedModel",
  "idInjection": true,
  "properties": {
    "title": {
      "type": "string",
      "required": true
    },
    "desc": {
      "type": "string"
    },
    "created": {
      "type": "date",
      "required": true
    }
  },
  "validations": [],
  "relations": {
    "author": {
      "type": "belongsTo",
      "model": "Account",
      "foreignKey": "authorId"
    }
  },
  "acls": [],
  "methods": []
}
```
Note that property `created` is required.

When I try to save a new object with invalid `created` property, the object is successfully saved & stored value for that property is `null`.

Here is a screenshot from api explorer with a post operation https://www.dropbox.com/s/cbdlyb0a6uv51au/save-invalid-date.png

Is it a bug?

**Comment by raymondfeng** [comment]
I recently found that `new Date('invalid)` doesn't throw in JS and it creates an instance of Date. Which database do you use?

**Comment by aantipov** [comment]
mongodb

**Comment by bajtos** [comment]
When serializing to JSON, invalid dates are converted to `null`.

I think we should fix loopback-datasource-juggler to reject invalid date values.

**Comment by supasate** [comment]
@bajtos I tried to create with an invalid date value and it already throws `Error: Invalid date: <invalid value>`.

It seems like this issue has been solved. I see the code that handles this issue is located at
https://github.com/strongloop/loopback-datasource-juggler/blob/06cac9e7d4d05983e7a0646bb35bdfdf5340d15c/lib/model-builder.js#L527-L533.

Therefore, I think adding a test for the Date type in `datatype.test.js` of loopback-datasource-juggler might be enough for this issue.
